test/parallel/test-child-process-flush-stdio.js

```diff
@@ -20,10 +21,42 @@ function spawnWithReadable() {
     assert.strictEqual(code, 0);
     assert.strictEqual(signal, null);
     assert.strictEqual(Buffer.concat(buffer).toString().trim(), '123');
+    spawnWithPipe();
   }));
   p.stdout.on('readable', function() {
-    let buf;
-    while (buf = this.read())
-      buffer.push(buf);
+    setImmediate(() => {
+      let buf;
+      while (buf = this.read())
+        buffer.push(buf);
+    });
```

Comment:
This testcase was added to cover the issue fixed here: https://github.com/resin-io-playground/node/commit/12274a5c1b0a85f6f8eaf3f6e5c78b1de7617dfc

But the testcase passes even if the change of that commit is reverted. So I just fixed the testcase to reflect what was actually fixed there.